docs/2018-11-12-Unification.md

```diff
@@ -0,0 +1,360 @@
+Unification
+===========
+
+Background
+----------
+
+Whenever we want to apply an axiom `∀ Y . α(Y) -> •β(Y)` to a pattern `φ(X)`
+we have to find the pattern that contains whatever is common to `α(Y)`
+and `φ(X)` (i.e. their intersection, a.k.a. `φ(X) ∧ α(Y)`) and use that,
+together with `β(Y)` to compute a pattern `φ'(X)` such that `φ(X) -> •φ'(X)`.
+
+In practice, we want to find a term `t(x)`, preferably composed out of
+variables, constructors, functions and similar things,
+a substitution `S(X, Y)` and a predicate `P(X)` such that
+`φ(X) ∧ α(Y) = t(X) ∧ S(X, Y) ∧ P(X)`.
+
+It is debatable whether we want to have substitutions for the `X` variables.
+
+Problem
+-------
+
+* Finding if there are `t(X)`, `S(X, Y)` and `P(X)` such that
+  `φ(X) ∧ α(Y) = t(X) ∧ S(X, Y) ∧ P(X)`, i.e. finding if `φ(X)` and `α(Y)`
+  *unify*.
+* Actually finding those patterns.
+
+Additional constraints and assumptions
+--------------------------------------
+
+* The unification algorithm assumes that it mostly unifies terms.
```

Comment:
"mostly unifies terms" as in, term patterns? (As in patterns composed of functional symbols and variables, with no logical connectives)

Reply:
This PR includes a glossary, and I meant what is written there, so if there is any issue with the meaning of "term" or whether I should say "term pattern" instead, we should have it there.

You might be right that I should call it "term pattern" instead of just "term" (though we seem to have no issues with overloading, say, "function"), so I'll rename it if you say so.

Reply:
Yeah, IMO it should always be "term pattern" as "term" is too common a word on its own.

Reply:
Okay, done.